### PR TITLE
HSEARCH-3308 + HSEARCH-3356 + HSEARCH-3996 + HSEARCH-4052 Logs/exception fixes and polishing

### PR DESCRIPTION
**Pull request by yrodiere**
* [HSEARCH-4052](https://hibernate.atlassian.net/browse/HSEARCH-4052): Add missing properties such as "directory_provider" to HibernateSearch5Properties
* [HSEARCH-3996](https://hibernate.atlassian.net/browse/HSEARCH-3996): Double check all Log file offsets
* [HSEARCH-3356](https://hibernate.atlassian.net/browse/HSEARCH-3356): Use consistent phrasing in log messages and exceptions
* [HSEARCH-3308](https://hibernate.atlassian.net/browse/HSEARCH-3308): Check Search 6 log/exception messages and use Search 5 messages instead (with Search 5 ID) where possible

### CODE REVIEW

**Comment by yrodiere** [comment]
Putting on hold until #2423 gets merged.

**Comment by yrodiere** [comment]
I merged the PR this was depending on, and addressed Sonar/LGTM issues. Ready to review!
By the way, some commits in this PR are quite long and low-value... in particular " HSEARCH-3356 Use clear and consistent phrasing in log messages and exceptions". Feel free to skip reviewing these.

**Review by fax4ever — APPROVED** [review]
Very nice job. I like the new messages.
I've few comments, most of which are about the passive form using a subject derived from an object with a preposition, so feel free to ignore them.
+1 to merge

File: integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedBaseIT.java
Comment (yrodiere) [reply integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedBaseIT.java:457]: It's a completely valid solution when you're in this situation:

```
@Entity
@Indexed
public class MyEntity {
  // ...
}

@MapperSuperclass
public classs MySuperclass extends MyEntity {
  // ...
}

@Entity
public class MySubEntity extends MySuperclass {
  // ...
}
```

When you have this hierarchy, `MySuperclass` must be abstract, and if you don't want to make it abstract you need to annotate it with `@Indexed(enabled = false)`.

Since it's very specific I'd rather not go into details in the exception message, but I think users should be aware that `@Indexed(enabled = false)` exists.

**Comment by yrodiere** [comment]
Thanks. I updated the messages to address your comments. Let's merge once the CI build passes.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**
